### PR TITLE
Add users filter

### PR DESCRIPTION
**Pull request by wilpow-github**
### Context

As a TRA helpdesk agent
I want to filter users by TRN lookup status
So that I can easily identity which users need their TRN finding manually

### Changes proposed in this pull request

Added a filter to the LHS of the users page.
Added filtering by LookupStatus

### Checklist

-   [x] Attach to Trello card
-   [ ] Rebased master
-   [ ] Cleaned commit history
-   [ ] Tested by running locally
-   [ ] Reminder created to manually clean any removed app settings post deployment

### CODE REVIEW

**Comment by wilpow-github** [comment]
Screenshot 2023-02-07 181203
Screenshot 2023-02-07 181156

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/Users.cshtml [hunk 1]
```diff
@@ -4,109 +4,107 @@
...
-                        <govuk-pagination-ellipsis />
-                    }
+                            if (i > 0 && Model.PaginationPages[i - 1] != item - 1)
+                            {
+                                <govuk-pagination-ellipsis />
+                            }
 
-                    <govuk-pagination-item asp-page="Users" asp-route-pageNumber="@item" is-current="@(item == Model.PageNumber)" asp-fragment="closed-tab">@item</govuk-pagination-item>
-                }
+                            <govuk-pagination-item asp-page="Users" asp-all-route-data=@Model.GetPage(item) is-current="@(item == Model.PageNumber)">@item</govuk-pagination-item>
```
Comment (wilpow-github) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/Users.cshtml:56]: Apparently you can't pass "page" in via asp-all-route-data. Throws a null pointer exception which had me confused for a while

File: dotnet-authserver/src/TeacherIdentity.AuthServer/wwwroot/Styles/site.scss [hunk 2]
```diff
@@ -87,3 +87,72 @@ $app-button-inverse-hover-background-colour: govuk-tint($app-button-inverse-fore
...
+
+.moj-filter-layout {
+  @include govuk-clearfix;
+}
+
+.moj-filter-layout__filter {
+  box-shadow: inset 0 0 0 1px govuk-colour("light-grey"); // Extends the inset border left full height of the filters on mobile
+
+  @include govuk-media-query($until: desktop) {
+    margin-bottom: 20px;
```
Comment (wilpow-github) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/wwwroot/Styles/site.scss:99]: I added this to allow some space below the filter on mobile. Everything else was copied over

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/Users.cshtml.cs [hunk 3]
```diff
@@ -61,24 +63,14 @@ public async Task<IActionResult> OnGet()
...
 
-        Open = await sortedUsers.Where(openPredicate).Select(mapUserInfo).ToArrayAsync();
-        OpenTotal = Open.Length;
-
-        Closed = await sortedUsers.Where(closedPredicate)
+        FilteredUsers = await sortedUsers.Where(filterPredicate)
             .Skip((PageNumber.Value - 1) * pageSize)
             .Take(pageSize)
             .Select(mapUserInfo)
             .ToArrayAsync();
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/Users.cshtml.cs:71]: Would be good to retain this 'out of range' page check

**Review by gunndabad — APPROVED** [review]

**Pull request merged**